src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.ASN1.Nid.cs

```diff
@@ -2,15 +2,42 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Collections.Concurrent;
+using System.Diagnostics;
 using System.Runtime.InteropServices;
+using System.Security.Cryptography;
 
 internal static partial class Interop
 {
     internal static partial class Crypto
     {
+        private static readonly ConcurrentDictionary<string, int> s_nidLookup =
```

Comment:
where does the string come from? Is it possible for malicious user to somehow fill this up?

Reply:
OID literals.  Since the OID comes from specs and the NID is just OpenSSL's identifier, I'm more comfortable hard-coding the OID